packages/react-reconciler/src/ReactFiberWorkLoop.new.js

```diff
@@ -2903,16 +2908,64 @@ function flushPassiveUnmountEffectsInsideOfDeletedTree(
       case ForwardRef:
       case SimpleMemoComponent:
       case Block: {
-        flushPassiveUnmountEffectsImpl(fiberToDelete, HookPassive);
+        flushPassiveUnmountEffectsForUnmountedFiberImpl(
+          fiberToDelete,
+          nearestMountedAncestor,
+        );
       }
     }
   }
 }
 
+function flushPassiveUnmountEffectsForUnmountedFiberImpl(
+  fiber: Fiber,
+  nearestMountedAncestor: Fiber,
```

Comment:
As much as I like to not DRY when things won't necessarily have the same concept going forward. This might be an unnecessary duplication.

Instead, you can just always pass the ancestor - in all the cases that call the equivalent non-unmounted case - even when that is the direct parent.

Reply:
That's what I did initially and Andrew's suggestion was to split them up. :smile: